### PR TITLE
Disallow the web profiler in the `robots.txt` file

### LINKED ISSUE #6592: Contao Crawler also checks profiler pages
### Affected version(s)

4.13

### Description

With activated Debug Mode, the Contao crawler (started via backend) also checks the symfony profiler pages (typical URI `https://abc.localhost/_profiler/24d1b4?panel=request`).

The solution is simple: _Don't activate the Debug Mode when crawling!_

But maybe a fix is as simple as adding a `<!-- indexer::stop -->` before the synfony panel (and continue the indexer after the panel).

### PR DESCRIPTION
**Pull request by aschempp**
Fixes https://github.com/contao/contao/issues/6592

Technically, the `/_profiler/` and `/_wdt` routes can be reconfigured, but I don't think we can _detect_ that and it's probably not worth the effort 🙃

### CODE REVIEW

**Review by bytehead — DISMISSED** [review]

**Review by Toflar — DISMISSED** [review]

File: core-bundle/src/EventListener/RobotsTxtListener.php [hunk 1]
```diff
@@ -60,6 +63,11 @@ public function __invoke(RobotsTxtEvent $event): void
             $directiveList = $record->getDirectiveList();
             $directiveList->add(new Directive('Disallow', $this->routePrefix.'/'));
             $directiveList->add(new Directive('Disallow', '/_contao/'));
+
+            if ($this->webDebugToolbarListener && $this->webDebugToolbarListener->isEnabled()) {
+                $directiveList->add(new Directive('Disallow', '/_profiler/'));
+                $directiveList->add(new Directive('Disallow', '/_wdt/'));
+            }
```
Comment (leofeyer) [inline core-bundle/src/EventListener/RobotsTxtListener.php:70]:
```suggestion
            $directiveList->add(new Directive('Disallow', '/_profiler/'));
            $directiveList->add(new Directive('Disallow', '/_wdt/'));
```

Do we really need to check whether the toolbar is enabled? I doubt that anyone uses these routes for their own app in production, so we might as well always disallow them.

Comment (Toflar) [reply core-bundle/src/EventListener/RobotsTxtListener.php:70]: Which would make them public in all the `robots.txt` responses of all Contao setups. I too think that it's better like it is now to only add those in debug mode. Feels cleaner.

Comment (leofeyer) [reply core-bundle/src/EventListener/RobotsTxtListener.php:70]: Okay, fine with me.

**Comment by aschempp** [comment]
ich weiss leider nicht wie ich den RequireChecker fixen kann …

**Comment by ausi** [comment]
> ich weiss leider nicht wie ich den RequireChecker fixen kann …

by adding the affected symbol to this list: https://github.com/contao/contao/blob/4.13/tools/require-checker/config.json

**Comment by aschempp** [comment]
So that will ignore all use of that class in the future? Shouldn't it notice the dependency is optional?

**Comment by ausi** [comment]
> So that will ignore all use of that class in the future?

Correct.

> Shouldn't it notice the dependency is optional?

AFAIK `maglnet/composer-require-checker` is a pretty simple tool that just looks for all symbols that are used. It doesn’t analyze codepaths to detect optional dependencies. Symbols for optional dependencies need to be excluded manually.

**Review by leofeyer — APPROVED** [review]

**Comment by leofeyer** [comment]
Thank you @aschempp.

**Pull request merged**